### PR TITLE
feat: [ENG-1786] add the ConsumeQueue to the poisonQueueHandler to allow consuming the poison queue

### PR DESCRIPTION
**Pull request by phturb**
Add the logic to consume the poison queue
This logic makes sure to take all the result up to the latest offset when the consumption is done.
The consumption record count is non-deterministic because we are consuming by batch the records and we complete when the new batch includes the final offset define when the consumption started. We can end up with a small amount of records on top of the final offset, but the logic makes sure that we stop when we include that offset.
The logic intend to make a single execution with all the records to the handler. This is to prevent scenario where we can push back more event on the queue and accidentally consume them.

### CODE REVIEW

File: pubsubx/kgox/admin_client_impl_test.go [hunk 1]
```diff
@@ -30,87 +30,107 @@ func TestKadminClient_DeleteTopicsWithRetryTopics(t *testing.T) {
...
-			defer kadmCl.Close()
-			kadmCl.DeleteTopics(context.Background(), topics...)
-		})
 		require.NoError(t, err)
-		assert.Contains(t, createdTopics, topics[0])
-		assert.Contains(t, createdTopics, topics[1])
+		for _, top := range topics {
+			assert.Contains(t, createdTopics, top)
+		}
+		time.Sleep(time.Second)
```
Comment (pobeaulieu1) [inline pubsubx/kgox/admin_client_impl_test.go:63]: polling with a timeout

File: pubsubx/kgox/poison_queue_handler.go [hunk 2]
```diff
@@ -116,5 +120,105 @@ func (pqh *poisonQueueHandler) generatePoisonQueueRecord(ctx context.Context, to
...
+	msgs := make([]*messagex.Message, 0)
+	errs := make([]error, 0)
+FLOOP:
+	for {
+		select {
+		case <-cctx.Done():
+			break FLOOP
+		default:
+		}
+		tcctx, tcancel := context.WithTimeout(cctx, 5*time.Second)
```
Comment (pobeaulieu1) [inline pubsubx/kgox/poison_queue_handler.go:161]: we could put the poll timeout in a constant.

File: pubsubx/kgox/poison_queue_handler.go [hunk 3]
```diff
@@ -116,5 +120,105 @@ func (pqh *poisonQueueHandler) generatePoisonQueueRecord(ctx context.Context, to
...
+		// Check if it's a FetchErr, by default PollRecords return a single fetch with a single topic and single partition holding the err
+		if len(fetches) == 1 &&
+			len(fetches[0].Topics) == 1 &&
+			len(fetches[0].Topics[0].Partitions) == 1 &&
+			fetches[0].Topics[0].Partitions[0].Err != nil {
+			if fetches[0].Topics[0].Partitions[0].Err != tcctx.Err() {
+				err = fetches[0].Topics[0].Partitions[0].Err
+				pqh.l.WithError(err).Errorf("error fetches returned")
+			} else {
+				pqh.l.WithError(fetches[0].Topics[0].Partitions[0].Err).Infof("expected fetche error trigger consumption termination")
```
Comment (pobeaulieu1) [inline pubsubx/kgox/poison_queue_handler.go:172]:
```suggestion
				pqh.l.WithError(fetches[0].Topics[0].Partitions[0].Err).Infof("expected fetch error trigger consumption termination")
```

File: pubsubx/kgox/poison_queue_handler.go (hunk 2 above)
Comment (pobeaulieu1) [reply pubsubx/kgox/poison_queue_handler.go:161]: Creating a new context for each iteration might be inefficient sinbce it introduces overhead.
we could use a single context.WithTimeout for the entire loop.

Comment (pobeaulieu1) [reply pubsubx/kgox/poison_queue_handler.go:161]: Also if it can be an additional mechanism to ensure we are not stuck in an infine loop.

**Comment by pascalesy** [comment]
This is not directly linked to this PR but as we use a default message max byte size of 10MB for our topics I think we should add a test for a poison event with a size >1MB. At first glance, I believe the current implementation is fine but this is an edge case that is easy to forget.

File: pubsubx/kgox/poison_queue_handler.go (hunk 2 above)
Comment (phturb) [reply pubsubx/kgox/poison_queue_handler.go:161]: The timeout needs to be per loop, because if it takes 4 seconds to empty the buffer and then it would take 4 second again to empty the second loop exeuction, we would timeout and never be able to retrieve all the entries.
The 4 seconds timeout is to end the loop when we get to the end of the queue. The poll is a blocking execution so without the timeout we would be blocking the code indefinitely.
(I tested it in a previous attempt)

**Comment by phturb** [comment]
> This is not directly linked to this PR but as we use a default message max byte size of 10MB for our topics I think we should add a test for a poison event with a size >1MB. At first glance, I believe the current implementation is fine but this is an edge case that is easy to forget.

https://linear.app/clinia/issue/ENG-1808/add-test-and-guardrail-for-poison-queue-events-that-are-encapsulating

File: pubsubx/kgox/poison_queue_handler.go [hunk 4]
```diff
@@ -116,5 +121,105 @@ func (pqh *poisonQueueHandler) generatePoisonQueueRecord(ctx context.Context, to
...
+		}
+		if len(fetches) == 0 {
+			pqh.l.Infof("no fetches were found, assuming poison queue is empty")
+			tcancel()
+			break FLOOP
+		}
+
+		for _, f := range fetches {
+			for _, t := range f.Topics {
+				for _, p := range t.Partitions {
```
Comment (pascalesy) [inline pubsubx/kgox/poison_queue_handler.go:186]: lets validate `endOffsets[t.Topic]` and `endOffsets[t.Topic][p.Partition]` is found for safety
